src/content-tags-drawer/ContentTagsCollapsible.test.jsx

```diff
@@ -200,6 +239,66 @@ describe('<ContentTagsCollapsible />', () => {
     });
   });
 
+  it('should handle search term change', async () => {
+    jest.useFakeTimers(); // To account for debounce timer
+
+    await act(async () => {
+      const {
+        container,
+        getByRole,
+        getByText,
+        getByDisplayValue,
+      } = render(
+        <ContentTagsCollapsibleComponent
+          contentId={data.contentId}
+          taxonomyAndTagsData={data.taxonomyAndTagsData}
+          editable={data.editable}
+        />,
+      );
+
+      // Expand the Taxonomy to view applied tags and "Add tags" button
+      const expandToggle = container.getElementsByClassName('collapsible-trigger')[0];
+      await act(async () => {
+        fireEvent.click(expandToggle);
+      });
+
+      // Click on "Add tags" button to open dropdown
+      const addTagsButton = getByText(messages.addTagsButtonText.defaultMessage);
+      await act(async () => {
+        fireEvent.click(addTagsButton);
+      });
+
+      // Get the search field
+      const searchField = getByRole('searchbox');
+
+      const searchTerm = 'memo';
+
+      // Trigger a change in the search field
+      await userEvent.type(searchField, searchTerm);
```

Comment:
This was alot more trouble than I hoped, it just wouldn't work if I put it inside the `act`. I spent a significant amount of time trying to get it to work.

I was facing a similar issue to what was mentioned in https://github.com/testing-library/user-event/issues/539. Add the `delay: 1` fixes it but I then start getting warnings in the console about `act` being calling before a previous one is terminated. I started changing the structure and adding more things here and there to fix it, but it felt too hacky/messy so it was cleaner to leave it as is.

All this code is inside an overlying `act` at the beginning of the testcase (I tried removing thinking it would solve the issues I faced, but it didn't.), so technically it is already inside an `act` if that was the main concern.

Reply:
It might be that the userevents library is also wrapping things in act already like fireEvent. In that case we can skip act for all those calls as well.

Reply:
Yup, that worked. I've updated the tests to remove the unnecessary `act` calls, mainly around `fireEvent`. Its ready to go.